src/diff-snapshot.js

```diff
@@ -45,13 +100,16 @@ function diffImageToSnapshot(options) {
 
     const diffConfig = Object.assign({}, defaultDiffConfig, customDiffConfig);
 
-    const receivedImage = PNG.sync.read(receivedImageBuffer);
-    const baselineImage = PNG.sync.read(fs.readFileSync(baselineSnapshotPath));
-
-    if (
-      receivedImage.height !== baselineImage.height || receivedImage.width !== baselineImage.width
-    ) {
-      throw new Error('toMatchImageSnapshot(): Received image size must match baseline snapshot size in order to make comparison.');
+    let receivedImage = PNG.sync.read(receivedImageBuffer);
+    let baselineImage = PNG.sync.read(fs.readFileSync(baselineSnapshotPath));
+    const hasSizeMismatch = (
+      receivedImage.height !== baselineImage.height ||
+      receivedImage.width !== baselineImage.width
+    );
+    // Align images in size if different
+    if (hasSizeMismatch) {
+      const resizedImages = alignImagesToSameSize(receivedImage, baselineImage);
```

Comment:
Key change, this resolves both images in same size **and** with new filled area after resize